serialization/compact_config.go

```diff
@@ -51,3 +63,55 @@ func (cc *CompactConfig) ensureSerializers() {
 		cc.serializers = make(map[string]CompactSerializer)
 	}
 }
+
+func (cc *CompactConfig) checkNoDefaultSerializer() error {
+	// note that the check we are doing is just information.
+	// the default serializers are looked uo before the compact serializer,
+	// so the user cannot override any default serializer.
+	var cr CompactReader
+	var cw CompactWriter
+	var ds IdentifiedDataSerializable
+	var p Portable
+	var b *big.Int
+	ss := []reflect.Type{
+		reflect.TypeOf(cr),
+		reflect.TypeOf(cw),
+		reflect.TypeOf(ds),
+		reflect.TypeOf(p),
+		reflect.TypeOf(byte(0)),
+		reflect.TypeOf(false),
+		reflect.TypeOf(""),
+		reflect.TypeOf(uint16(0)),
+		reflect.TypeOf(int16(0)),
+		reflect.TypeOf(int32(0)),
+		reflect.TypeOf(int64(0)),
+		reflect.TypeOf(float32(0)),
+		reflect.TypeOf(float64(0)),
+		reflect.TypeOf([]bool{}),
+		reflect.TypeOf([]string{}),
+		reflect.TypeOf([]byte{}),
+		reflect.TypeOf([]uint16{}),
+		reflect.TypeOf([]int16{}),
+		reflect.TypeOf([]float32{}),
+		reflect.TypeOf([]float64{}),
+		reflect.TypeOf(types.UUID{}),
+		reflect.TypeOf(b),
+		reflect.TypeOf(types.Decimal{}),
+		reflect.TypeOf(JSON("")),
+		reflect.TypeOf([]interface{}{}),
+		reflect.TypeOf(types.LocalDate{}),
+		reflect.TypeOf(types.LocalTime{}),
+		reflect.TypeOf(types.LocalDateTime{}),
+		reflect.TypeOf(types.OffsetDateTime{}),
```

Comment:
this might be fine for now, but I believe we can easily forget to update this list, when we add a new type to support to serialization service. is there a way to do it there without defining these here?

Reply:
I've added a note at: https://github.com/hazelcast/hazelcast-go-client/pull/942/commits/d6776c52c4a31420deea2b005bd514e0e35e5e7b